### PR TITLE
hooks: pre-push: Add tag format check

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: .hooks/pre-push [hunk 1]
```diff
@@ -4,6 +4,21 @@ set -e
 
 echo "Running pre push hook!"
 repository_path=$(git rev-parse --show-toplevel)
+tag_name=$(git tag --points-at=HEAD | head -n 1)
+
+[ ! -z "$tag_name" ] && {
```
Comment (joaoantoniocardoso) [inline .hooks/pre-push:9]:
```suggestion
[ -n "$tag_name" ] && {
```
isn't it simpler?

File: .hooks/pre-push [hunk 2]
```diff
@@ -4,6 +4,21 @@ set -e
...
 repository_path=$(git rev-parse --show-toplevel)
+tag_name=$(git tag --points-at=HEAD | head -n 1)
+
+[ -n "$tag_name" ] && {
+    echo "Current reference has the following tag: ${tag_name}"
+    dialog --defaultno --yesno "Are you pushing a tag ?" 20 60 && {
+        echo "Checking tag name.."
+        # Check if tag follows our semver convention
+        # MAJOR.MINOR.PATCH or MAJOR.MINOR.PATCH-SOMETHING.VERSION
+        $(echo "$tag_name" | grep -Po "^\d+\.\d+\.\d+(\-[A-z]+\.\d+)?$|^\d+\.\d+\.\d+$" > /dev/null) || {
```
Comment (joaoantoniocardoso) [inline .hooks/pre-push:15]: Consider [this one](https://regexr.com/3bkmj).

Comment (patrickelectric) [reply .hooks/pre-push:15]: I'm not willing to maintain this option, at least what we currently have is easy and we can understand.

Comment (joaoantoniocardoso) [reply .hooks/pre-push:15]: and untested, but ok

**Review by joaoantoniocardoso — APPROVED** [review]

**Pull request merged**